### PR TITLE
Database primitives

### PR DESCRIPTION
**Pull request by AlexJones2515**

### CODE REVIEW

**Comment by jmccook96** [comment]
Nice stuff, but I hadn't considered if we wanted to expose ID's to the systems interfacing.... That's an interesting thought experiment. We might need to deliberate on that a little.

Like I don't know if there will be scenarios in which we will want to interact via index/id that isn't internally inside of DataObjects. Although I also don't know if there's any harm either.

Thing to consider is when making a new object for us to store, its not the object or the objects creators job to assign it an ID, that has to happen inside of the DAO- which makes an inconsistency in the data held vs the data used...

**Comment by AlexJones2515** [comment]
Would you like the removal of the entering of the ID's then?

**Comment by jmccook96** [comment]
> Would you like the removal of the entering of the ID's then?

I don't know lol. Deal with the conflicts fist, and I'll approve for the time being.  Maybe cut it from the constructors, but leave it functionally present otherwise, and we'll do more trimmings once we have a better thought of it.

**Review by AlexJones2515 — COMMENTED** [review]
?

File: src/main/java/com/bookclub/model/Book.java [hunk 1]
```diff
@@ -1,84 +1,44 @@
...
-     * Both title and author must be non-null and non-empty.
-     *
-     * @param title  the title of the book
-     * @param author the author of the book
-     * @throws IllegalArgumentException if the title or author is null or empty
-     */
-    public Book(String title, String author) {
-        setTitle(title);
-        setAuthor(author);
+    public Book(Integer bookid, String title, String author, String genre) {
```
Comment (jmccook96) [inline src/main/java/com/bookclub/model/Book.java:10]: By my comment I meant for constructors like this, we don't need to worry about passing in all the content that could be optional (null) during construction as we can add these later. 

Would also be good to not have to worry about ID whenever we make something as it'll never be correct unless it comes from the DAO. So unforunately I'm asking to essentially revert the change to the constructor back to the original one.

```suggestion
    public Book(String title, String author) {
```

File: src/main/java/com/bookclub/model/Book.java [hunk 2]
```diff
@@ -1,84 +1,44 @@
...
- * Both title and author must be non-null and non-empty.
- */
 public class Book {
 
     private String title;
     private String author;
+    private Integer bookid;
+    private String genre;
 
-    /**
```
Comment (jmccook96) [inline src/main/java/com/bookclub/model/Book.java:13]: I know it's a pain but could I also ask to return all the other functionality and comments that you are trimming with this push please :)

File: src/main/java/com/bookclub/model/Event.java [hunk 3]
```diff
@@ -6,18 +6,30 @@ public class Event {
     private String organizer;
     private String date;
     private String location;
+    private Integer eventid;
+    private String description;
 
-    public Event(String title, String organizer, String date, String location) {
+    public Event(Integer eventid, String title, String organizer, String date, String location, String description) {
```
Comment (jmccook96) [inline src/main/java/com/bookclub/model/Event.java:12]: Same here, treat ID and description as optional for the time being, so not required when constructing through the constructor we have 😅 

```suggestion
    public Event(String title, String organizer, String date, String location) {
```

File: src/main/java/com/bookclub/model/Book.java (hunk 2 above)
Comment (AlexJones2515) [reply src/main/java/com/bookclub/model/Book.java:13]: I don't even remember actively removing this, will put it back in

**Pull request closed without merging**